src/python/pants/backend/python/tasks/python_repl.py

```diff
@@ -48,6 +49,11 @@ def setup_repl_session(self, targets):
       entry_point = 'code:interact'
     pex_info = PexInfo.default()
     pex_info.entry_point = entry_point
+    python_setup = PythonSetup.global_instance()
+    if python_setup.get_options().is_flagged('interpreter_constraints'):
```

Comment:
I believe that this is _also_ not necessary if the `is_flagged` logic moves into `compatibility_or_constraints`, because `python_execution_task_base.py` will do the right thing.